src/sources/internal_metrics.rs

```diff
@@ -9,23 +9,12 @@ use futures::{stream, SinkExt, StreamExt};
 use serde::{Deserialize, Serialize};
 use tokio::time;
 
-#[serde(deny_unknown_fields)]
-#[derive(Deserialize, Serialize, Debug, Clone)]
+#[derive(Deserialize, Serialize, Debug, Clone, Derivative)]
+#[derivative(Default)]
+#[serde(deny_unknown_fields, default)]
 pub struct InternalMetricsConfig {
-    #[serde(default = "default_scrape_interval_secs")]
-    scrape_interval_secs: u64,
-}
-
-pub const fn default_scrape_interval_secs() -> u64 {
-    2
-}
-
-impl Default for InternalMetricsConfig {
-    fn default() -> Self {
-        Self {
-            scrape_interval_secs: default_scrape_interval_secs(),
-        }
-    }
+    #[derivative(Default(value = "2.0"))]
+    scrape_interval_secs: f64,
```

Comment:
I think this change to `f64` is ok, but we may want to update the `scrape_interval_secs` on other components (like the `prometheus_scrape` source) to be `f64`s to avoid confusion. Alternatively, we put this back to `u64` with a lower-bound of `1` (which we may also want to roll out to other `scrape_interval_secs`).

Reply:
Let's change everything to `f64` - it makes way more sense to specify seconds with floats rather than integers.

Reply:
Don't want to do in for all of the components in this PR though. Would the resolution be to revert to seconds here and revisit it in a separate PR that addresses just that? Or should we let this one pass as `f64`?

Reply:
:+1: I'm fine with switching them all. My preference would be to leave this as `u64` and have a separate PR that changes them all at once.

Reply:
Switched to `u64`.